pyccl/nonlimber_FKEM.py

```diff
@@ -0,0 +1,234 @@
+__all__ = ("nonlimber_FKEM",)
+"""Written by Paul Rogozenski (paulrogozenski@arizona.edu),
+ implementing the FKEM non-limber integration method of the N5K challenge
+ detailed in this paper: https://arxiv.org/pdf/1911.11947.pdf .
+We utilize a modified generalized version of FFTLog
+ (https://jila.colorado.edu/~ajsh/FFTLog/fftlog.pdf)
+ to compute integrals over spherical bessel functions
+"""
+import numpy as np
+from . import lib, check
+from .pyutils import integ_types
+from scipy.interpolate import interp1d
+from pyccl.pyutils import _fftlog_transform_general
+import pyccl as ccl
+
+
```

Comment:
One general comment/question before I continue reviewing: since (presumably) we may implement another non-limber method in the future, and the code for angular C_ells now lives in two different files, would it make sense to put all the C_ell code into its own submodule for tidyness and encapsulation? Otherwise I find it weird to have these two files lying around instead of just putting all the code in `cells.py`.

Reply:
Associated comment: all functions/methods not starting with `_` are considered public and part of the API, and hence should be properly documented. In this sense, the method below (`get_general_params`) is very specific of FKEM, and the name would suggest otherwise to an external user (especially since it's not documented).